### PR TITLE
Datadog Events Exporter

### PR DESCRIPTION
**Pull request by fdpeiter**
Adding simple exporter to send messages as Datadog Events.

### CODE REVIEW

**Comment by justinbarrick** [comment]
Welcome @fdpeiter! It looks like this is your first PR to justinbarrick/fluxcloud 🎉

**Comment by justinbarrick** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/justinbarrick/fluxcloud/pull/66#" title="Author self-approved">fdpeiter</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **justinbarrick**

If they are not already assigned, you can assign the PR to them by writing `/assign @justinbarrick` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=justinbarrick%2Ffluxcloud).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/justinbarrick/fluxcloud/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by justinbarrick** [comment]
Hi @fdpeiter. Thanks for your PR.

I'm waiting for a [justinbarrick](https://github.com/orgs/justinbarrick/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/justinbarrick/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=justinbarrick%2Ffluxcloud).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by marcosnils** [comment]
LGTM! :tada:

File: pkg/exporters/datadog.go [hunk 1]
```diff
@@ -49,12 +49,10 @@ func (d *Datadog) NewDatadogEvent(message msg.Message) []DatadogEvent {
...
 	for _, event := range message.Event.ServiceIDs {
 		tags := []string{"application:flux", "fluxEventType:" + message.Event.Type}
-		// If inside a Kubernetes cluster, add tags to specify the cluster and the namespace
-		_, kubernetes := os.LookupEnv("KUBERNETES_SERVICE_HOST")
-		if kubernetes {
-			namespace, _, _ := event.Components()
-			tags = append(tags, "namespace:"+namespace)
-		}
+		namespace, kind, name := event.Components()
+		tags = append(tags, "fluxnamespace:"+namespace)
```
Comment (marcosnils) [inline pkg/exporters/datadog.go:53]: Shall we add tests to validate that these tags are being added?

**Comment by nagypeterjob** [comment]
Hey guys, can I help anything so we can proceed with this PR?

File: pkg/exporters/datadog.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		}
+		events = append(events, DatadogEvent{message.Title, message.Body, tags})
+	}
+	return events
+}
+
+// Send uses the datadog api client to post the event
+func (d *Datadog) Send(c context.Context, client *http.Client, message msg.Message) error {
+	for _, event := range d.NewDatadogEvent(message) {
+		fmt.Println(event)
```
Comment (marcosnils) [inline pkg/exporters/datadog.go:70]: Remove print?

**Comment by nagypeterjob** [comment]
I would gladly finish the PR given the permissions to do so. Would it be okay, to lift the changes to a completely new fork and merge the PR after I have fixed your request?

Sorry for the haste but it would be stellar if I could use this feature without maintaining an own fork until the upstream PR is merged.

**Comment by marcosnils** [comment]
@nagypeterjob I work with @fdpeiter and we're planning to address the missing bits here. We'll be also adding some missing fields we've identified could be sent to DD as well.

**Comment by justinbarrick** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/justinbarrick/fluxcloud/pull/66#" title="Author self-approved">fdpeiter</a>*, *<a href="https://github.com/justinbarrick/fluxcloud/pull/66#pullrequestreview-431783424" title="Approved">marcosnils</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **justinbarrick**

If they are not already assigned, you can assign the PR to them by writing `/assign @justinbarrick` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=justinbarrick%2Ffluxcloud).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/justinbarrick/fluxcloud/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by marcosnils** [comment]
@justinbarrick this should be ready to merge. :bell:

**Comment by fdpeiter** [comment]
@nagypeterjob in the meantime, you can generate this Docker image using the following Dockerfile, which we already use in production:
```
FROM golang:1.12 as builder
WORKDIR /app
COPY . .
RUN GO111MODULE=on GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o ./fluxcloud ./cmd/

FROM alpine
RUN apk update && apk add ca-certificates

FROM gcr.io/distroless/static@sha256:48e0d165f07d499c02732d924e84efbc73df8021b12c24940e18a9306589430e
COPY --from=0 /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt

WORKDIR /app
COPY --from=builder /app/fluxcloud .
EXPOSE 3031
ENTRYPOINT ["/app/fluxcloud"]

```

**Comment by marcosnils** [comment]
We're continuing the development of his project here: https://github.com/topfreegames/fluxcloud Here's the reason for that https://github.com/justinbarrick/fluxcloud/issues/68

@fdpeiter you can close this issue.

**Pull request closed without merging**